extensions/functions_string.yaml

```diff
@@ -248,3 +248,192 @@ scalar_functions:
+            variadic:
+              min: 2
+          - value: "varchar<L2>"
+            name: "separator"
+            description: Character to separate strings by.
+        return: "varchar<L3>"
+  -
+    name: repeat
+    description: Repeat a string a specified number of times.
+    impls:
+      - args:
+          - value: "string"
+          - value: i64
+        return: "string"
+      - args:
+          - value: "varchar<L1>"
+          - value: i64
+        return: "varchar<L2>"
+  -
+    name: reverse
+    description: Returns the string in reverse order.
+    impls:
+      - args:
+          - value: "string"
+        return: "string"
+      - args:
+          - value: "varchar<L1>"
+        return: "varchar<L1>"
+      - args:
+          - value: "fixedchar<L1>"
+        return: "fixedchar<L1>"
+  -
+    name: replace_slice
+    description: >-
+      Replace a slice of the input string.  A specified 'length' of characters will be deleted from
+      the input string beginning at the 'start' position and will be replaced by a new string.  A
+      start value of 1 indicates the first character of the input string. If start is negative
+      or zero, or greater than the length of the input string, a null string is returned. If 'length'
+      is negative, a null string is returned.  If 'length' is zero, inserting of the new string
+      occurs at the start of the input string and no characters are deleted. If 'length' is
+      greater than the input string, deletion will occur up to the last character of the input string.
+    impls:
+      - args:
+          - value: "string"
+            name: "input"
+            description: Input string.
+          - value: i64
+            name: "start"
+            description: The position in the string to start deleting/inserting characters.
+          - value: i64
+            name: "length"
+            description: The number of characters to delete from the input string.
+          - value: "string"
+            name: "replacement"
+            description: The new string to insert at the start position.
+        return: "string"
+      - args:
+          - value: "varchar<L1>"
+            name: "input"
+            description: Input string.
+          - value: i64
+            name: "start"
+            description: The position in the string to start deleting/inserting characters.
+          - value: i64
+            name: "length"
+            description: The number of characters to delete from the input string.
+          - value: "varchar<L2>"
+            name: "replacement"
+            description: The new string to insert at the start position.
+        return: "varchar<L3>"
+  -
+    name: lower
+    description: >-
+      Transform the string to lower case characters. Implementation should follow the utf8_unicode_ci
+      collations according to the Unicode Collation Algorithm described at http://www.unicode.org/reports/tr10/.
```

Comment:
I was mainly looking here when writing this:
https://forums.mysql.com/read.php?103,187048,188748#msg-188748
https://dev.mysql.com/doc/refman/5.7/en/charset-unicode-sets.html

```
MySQL implements the xxx_unicode_ci collations according to the Unicode Collation Algorithm (UCA) described at http://www.unicode.org/reports/tr10/
```

Follow up to @jacques-n comment here:
https://github.com/substrait-io/substrait/pull/245#discussion_r921711088

Not sure if these are the kinds of rules we're looking for, so open to more discussion.

Reply:
I added the option, let me know if you think it's okay.

Reply:
I can't really fault you for this because in all likelihood nothing says anything about this, but I'd strongly prefer we use underscores for the option names. Dashes are an absolute PITA to parse in any DSL because they're ambiguous with subtractions. If there's precedent for dashes you can just leave it as is though, I guess...

Reply:
Thanks for the catch...I think all the other options actually do use underscores.  I just changed it.